Helm: release notes for Mimir 2.3 (#2775)
grafana/mimir: docs/sources/release-notes/v2.3.md

- **Added `backfill` command to mimirtool to upload Prometheus TSDB blocks:**
  The `mimirtool backfill` command will make the process of back-filling Prometheus TSDB blocks, using an API available in the compactor, even easier which allows for a migration to Grafana Mimir while keeping your previous Prometheus TSDB data.

> - **Helm Helm Chart improvements:** The Mimir Helm chart is the best way to install Mimir on Kubernetes. As part of the Mimir 2.3 release, we’re also releasing version 3.1 of the Helm chart. Notable enhancements follow. For the full list of changes, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).

Reviewer: ```suggestion
- **New Helm chart minor release:** The Mimir Helm chart is the best way to install Mimir on Kubernetes. As part of the Mimir 2.3 release, we’re also releasing version 3.1 of the Mimir Helm chart. Notable enhancements follow. For the full list of changes, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).
```

Reviewer: I think in a few more releases we can stop saying "The Mimir Helm chart is the best way to install Mimir on Kubernetes.", but for now let's build the muscle in the community.

>   - The MinIO sub chart dependency is upgraded from a deprecated Helm chart (https://helm.min.io/) to the supported chart's version (https://charts.min.io/). This includes a **breaking change** of how the administrator password is set, however as the _built-in_ MinIO is not recommended for production, this change did not warrant a new major version of the Mimir Helm chart.

Reviewer: ```suggestion
  - We've upgraded the MinIO subchart dependency from a [deprecated chart](https://helm.min.io/) to the [supported one](https://charts.min.io/). This creates a **breaking change** in how the administrator password is set. However, as the built-in MinIO is not a recommended object store for production use cases, this change did not warrant a new major version of the Mimir Helm chart.
```

>   - The backfill API endpoints are now exposed on the Nginx gateway.

Reviewer: ```suggestion
  - The backfill API endpoints for importing historic time series data are now exposed on the Nginx gateway.
```

>   - Default to injecting the `no_auth_tenant` from the Mimir configuration as the value for `X-Scope-OrgID` in Nginx instead of hard coded `anonymous`.

Reviewer 2: not feeling strongly about this
```suggestion
  - Default to injecting the `no_auth_tenant` from the Mimir configuration as the value for `X-Scope-OrgID` in Nginx instead of hard coded `anonymous`.
  - Memberlist now uses DNS service-discovery by default. This allows Mimir only join a subset of the memberlist cluster when starting. This makes startup in large clusters faster.
```

Author: yeah, I felt like that's more an implementation detail

Reviewer: - Memberlist now uses DNS service-discovery by default, which should decrease startup time for large Mimir clusters.
^I tried rewriting to remove some of the implementation detail and emphasize impact. 

I'm not sure how if at all I should change my suggestion based on @pracucci 's review comment though.

Reviewer: I also updated the language for the nginx callout to also try to call out the rationale for the change. 

  - Nginx now sets the value of the `X-Scope-OrgID` header equal to the value of Mimir's `no_auth_tenant` flag by default. The previous release had set `X-Scope-OrgID` to the hard coded value of `anonymous` by default which [complicated the process of migrating to Mimir](https://github.com/grafana/mimir/issues/2563).

Author: committing this but changing "flag" to "parameter" since we only call CLI flags "flags", the YAML contains parameters

>   - Default to injecting the `no_auth_tenant` from the Mimir configuration as the value for `X-Scope-OrgID` in Nginx instead of hard coded `anonymous`.

Reviewer 3: I would clarify that the default behaviour hasn't changed tho.

Author: tried to clarify in a suggestion bellow, wdty? @pracucci @09jvilla

  The `mimirtool backfill` command will make the process of back-filling Prometheus TSDB blocks, using an API available in the compactor, even easier which allows for a migration to Grafana Mimir while keeping your previous Prometheus TSDB data.

- **New Helm chart minor release:** The Mimir Helm chart is the best way to install Mimir on Kubernetes. As part of the Mimir 2.3 release, we’re also releasing version 3.1 of the Mimir Helm chart. Notable enhancements follow. For the full list of changes, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).
  - We've upgraded the MinIO subchart dependency from a [deprecated chart](https://helm.min.io/) to the [supported one](https://charts.min.io/). This creates a **breaking change** in how the administrator password is set. However, as the built-in MinIO is not a recommended object store for production use cases, this change did not warrant a new major version of the Mimir Helm chart.
  - The backfill API endpoints for importing historic time series data are now exposed on the Nginx gateway.

>   - Nginx now sets the value of the `X-Scope-OrgID` header equal to the value of Mimir's `no_auth_tenant` flag by default. The previous release had set the value of `X-Scope-OrgID` to `anonymous` by default which [complicated the process of migrating to Mimir](https://github.com/grafana/mimir/issues/2563). 

Author: ```suggestion
  - Nginx now sets the value of the `X-Scope-OrgID` header equal to the value of Mimir's `no_auth_tenant` parameter by default. The previous release had set the value of `X-Scope-OrgID` to `anonymous` by default which [complicated the process of migrating to Mimir](https://github.com/grafana/mimir/issues/2563). 
```

>   - Nginx now sets the value of the `X-Scope-OrgID` header equal to the value of Mimir's `no_auth_tenant` parameter by default. The previous release had set the value of `X-Scope-OrgID` to `anonymous` by default which [complicated the process of migrating to Mimir](https://github.com/grafana/mimir/issues/2563). 

Author: ```suggestion
  - Nginx now sets the value of the `X-Scope-OrgID` header equal to the value of Mimir's `no_auth_tenant` parameter by default. The previous release had set the value of `X-Scope-OrgID` to `anonymous` by default which [complicated the process of migrating to Mimir](https://github.com/grafana/mimir/issues/2563) when `no_auth_tenant` was set to a different value. If `no_auth_tenant` parameter was set to the default `anonymous`, then this change has no effect.
```

Reviewer: Sounds good!
